### PR TITLE
feat: load zkey from a file path, add file downloader in swift

### PR DESCRIPTION
**Pull request by vivianjeng**
Research for #65 

## TODO
- [ ] use `#[cfg(feature = "use_external_zkey")]` to control features
- [x] download and load arkzkey in `initializeMopro` (See 3.)
- [ ] download file in android and use it in proof
- [ ] benchmark different size of arkzkeys
- [ ] try on real device

1. 
load from a fixed path `e.g. "mopro-core/examples/circom/keccak256/target/keccak256_256_test_final.arkzkey"`
proving time in swift
```js
Time to deserialize proving key: 9.177727583s
Time to deserialize matrices: 70.397416ms
Initializing arkzkey took: 9.25s

Generating proof 2
Witness generation took: 423.89ms
Loading arkzkey took: 26.66ms

proof generation took: 1.36s
```
> there is no much difference between using `ARKZKEY` and `.arkzkey`
> Using `ARKZKEY` is even slightly slower than loading from a `.arkzkey` file

2. 
download from a server `e.g. "https://mopro.vivianjeng.xyz/keccak256_256_test_final.arkzkey"`
```js
Download ark key took: 0.021011948585510254

Time to deserialize proving key: 8.894824041s
Time to deserialize matrices: 26.038125ms
Initializing arkzkey took: 8.92s

Generating proof 2
Witness generation took: 410.84ms

// TODO: should be done in initialization
Time to deserialize proving key: 8.895085333s
Time to deserialize matrices: 24.071833ms
Loading arkzkey took: 8.94s
proof generation took: 979.47ms
```

3.
Init with arkzkey initialization
```js
Time to deserialize proving key: 8.930367541s
Time to deserialize matrices: 24.104291ms

Generating proof 2
Witness generation took: 429.28ms
Loading arkzkey took: 208.00ns
proof generation took: 963.43ms

Verification time 2: 3.227791ms
```

### CODE REVIEW

File: mopro-ios/MoproKit/Example/MoproKit/AnonAadhaarViewController.swift [hunk 1]
```diff
@@ -99,8 +103,12 @@ class AnonAadhaarViewController: UIViewController, WKScriptMessageHandler, WKNav
             // Record start time
             let start = CFAbsoluteTimeGetCurrent()
             
+            // Ark zkey url
+            let documentsUrl = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!
```
Comment (oskarth) [inline mopro-ios/MoproKit/Example/MoproKit/AnonAadhaarViewController.swift:107]: Is there some specific reason we are downloading via Swift as opposed to Rust? Seems like we'd have to implement same logic in many places, is it that much easier?

Comment (oskarth) [reply mopro-ios/MoproKit/Example/MoproKit/AnonAadhaarViewController.swift:107]: Interesting, no idea why!

Yes, that's a fair point.

Not sure which is best. If you got this to work let's start with this and in the future we can re-evaluate if needed!

**Comment by oskarth** [comment]
What's needed for this PR to be ready for review/merge? Can we gate this feature behind a feature flag maybe? So existing flow works but this can be toggled. Also basic docs in README or so would be great

File: mopro-core/src/middleware/circom/mod.rs [hunk 2]
```diff
@@ -169,6 +191,7 @@ pub fn witness_calculator() -> &'static Mutex<WitnessCalculator> {
 }
 
 pub fn generate_proof2(
+    zkey_path: &str,
```
[the remark this answers is not in the document]
Comment (vivianjeng) [reply mopro-core/src/middleware/circom/mod.rs:194]: oh this is used for experiment 
I am not going to keep this as input (will revert the change)

but `generate_proof_with_zkey` sounds good
I can include it in the new commit

**Comment by oskarth** [comment]
@vivianjeng can this PR be closed with #75 merged?

**Pull request closed without merging**